server/mem_agent/engine.py

```
@@ -0,0 +1,334 @@
+            orig_import = builtins.__import__
+
+            def custom_import(name, globals=None, locals=None, fromlist=(), level=0):
+                try:
+                    return orig_import(name, globals, locals, fromlist, level)
+                except ImportError as e:
+                    pkg = name.split(".")[0]
+                    logger.info(
+                        "Sandbox: attempting to install missing package '%s'", pkg
+                    )
+                    try:
+                        subprocess.run(
```

Ruff (0.14.1)

`subprocess` call: check for execution of untrusted input

(S603)